src/PostgREST/App.hs

```diff
@@ -403,8 +403,10 @@ handleInfo identifier RequestContext{..} =
     hasPK =
       not $ null $ tablePKCols ctxDbStructure (qiSchema identifier) (qiName identifier)
 
-handleInvoke :: InvokeMethod -> ProcDescription -> RequestContext -> DbHandler Wai.Response
-handleInvoke invMethod proc context@RequestContext{..} = do
+handleInvoke :: InvokeMethod -> Maybe ProcDescription -> RequestContext -> DbHandler Wai.Response
+handleInvoke invMethod maybeProc context@RequestContext{..} = do
+  proc <- maybe (throwError Error.RpcNotFound) return maybeProc
```

Comment:
I think it would be better to throw an `ApiRequestError` in `userApiRequest` instead and revert the change to `TargetProc` regarding the Maybe. That would be much cleaner.

Reply:
I agree, that may help to make better error messages too (solving the second bullet point of my first comment)